src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -64,6 +100,22 @@ class PromotionProvider(private val project: Project) {
         )
     }
 
+    private fun getSonarLintProjectNotifications(): SonarLintProjectNotifications {
+        return SonarLintUtils.getService(
+            project,
+            SonarLintProjectNotifications::class.java
+        )
+    }
+
+    private fun processFullProjectPromotion(notifications: SonarLintProjectNotifications) {
+        val firstAutoAnalysis = Instant.ofEpochMilli(PropertiesComponent.getInstance().getLong(FIRST_AUTO_ANALYSIS_DATE, 0L))
```

Comment:
We read the value twice in a row (here and just before calling this method), you could get the value only once and check at the same time